next.config.js

```diff
@@ -106,4 +107,31 @@ module.exports = withSourceMaps({
   assetPrefix: hasAssetPrefix ? `${scheme}://${process.env.ASSET_PREFIX}` : '',
   // Asset Prefix allows to use CDN for the generated js files
   // https://nextjs.org/docs/api-reference/next.config.js/cdn-support-with-asset-prefix
+  workboxOpts: {
+    swDest: process.env.NEXT_EXPORT
+      ? 'service-worker.js'
+      : 'static/service-worker.js',
+    runtimeCaching: [
+      {
+        urlPattern: /^https?.*/,
+        handler: 'NetworkFirst',
+        options: {
+          cacheName: 'offlineCache',
+          expiration: {
+            maxEntries: 200,
+          },
+        },
+      },
+    ],
+  },
+  async rewrites() {
+    return [
+      {
+        source: '/service-worker.js',
+        destination: '/service-worker.js',
+      },
+    ]
+  }
 });
```

Comment:
Where is the `service-worker.js` file? I cannot find it in this PR, so does my browser:
```
SW registration failed:  DOMException: 
Failed to register a ServiceWorker for scope ('https://planet-webapp-mwjjcg90l.vercel.app/') with script 
('https://planet-webapp-mwjjcg90l.vercel.app/service-worker.js'): 
The script has an unsupported MIME type ('text/html').
```